pkg/controllers/user/alert/deployer/upgradeimpl.go

```diff
@@ -383,3 +387,22 @@ func (l *AlertService) removeLegacyAlerting() error {
 	}
 	return nil
 }
+
+func (l *AlertService) removeFinalizerFromLegacyAlerting() error {
+	oldProjectAlert, err := l.oldProjectAlerts.List(metav1.ListOptions{})
```

Comment:
want to understand why we are not Lister here?

Reply:
same here. We should use lister instead.

Reply:
Because of upgrade first happening in the new version server set up, at this moment, the cache may not full sync, we may get nothing from Lister, then the range will be skipped. Using Lister may have some timing issue.